### PR TITLE
Refactor into monorepo

### PR DESCRIPTION
**Pull request by xyc**
### What does this PR do?
Refactor package into monorepo
- [x] separate into `plugin-templates` and `templates` packages 
- [x] update npm scripts: build, lint, test
- [x] fix tests
- [x] use i18n for `templates` package
- [x] clean up duplicate utils and messages in both packages

### What issues does this PR fix or reference?
@W-7746937@

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "version": "0.0.1",
```
Comment (lcampos) [inline lerna.json:2]: We'll need to change this to 49.x.x so we can keep the library and the plugin at the same version. A similar change would be needed on each module's package.json version field.

File: packages/templates/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "tslib": "^1",
+    "yeoman-environment": "2.4.0",
+    "yeoman-generator": "4.0.1"
+  },
+  "devDependencies": {
+    "@types/node": "^10",
+    "@types/yeoman-generator": "^3.1.3",
+    "shx": "0.2.2",
+    "tslint": "^6.1.0",
+    "typescript": "^3.8.3"
```
Comment (lcampos) [inline packages/templates/package.json:25]: Is this version of typescript compatible with the one we're using in the vscode repo ?

Comment (xyc) [reply packages/templates/package.json:25]: Good question, I didn't change the original version. Let me test it out.

File: lerna.json (hunk 1 above)
Comment (xyc) [reply lerna.json]: Updated all packages to 49.0.0

File: packages/templates/package.json (hunk 2 above)
Comment (xyc) [reply packages/templates/package.json:25]: Tested out on VS Code `npm pack`ed package and it looks good

**Pull request merged**